src/main/java/DemandeAvenantComponent.java

```diff
@@ -0,0 +1,200 @@
+					}
+				});
+			});
+		}
+	}
+
+	public void fillMontantAssureContratDansDemandeAvenantAcceptee(DossierPivot dossier, List<GarantiePivot> garantiesContrat) {
+		for (GarantiePivot garantieContrat : garantiesContrat) {
+			GarantiePivot garantieDemandeAvenantAcceptee = getGarantieDemandeAvenantAcceptee(dossier, garantieContrat);
+			if (garantieDemandeAvenantAcceptee != null && !StringUtils.equalsAny(garantieContrat.getCode(), "RE", "RC")) {
+				if (!garantieContrat.getCouvertures().isEmpty()) {
+					CouverturePivot couverture = garantieDemandeAvenantAcceptee.getCouvertures().get(0);
+					BigDecimal montantTotal = getMontantTotal(garantieContrat, garantieDemandeAvenantAcceptee);
+					couverture.setMontantAssure(montantTotal);
+					garantieDemandeAvenantAcceptee.setCouvertures(Collections.singletonList(couverture));
+				}
+			}
+		}
+	}
+
+	private GarantiePivot getGarantieDemandeAvenantAcceptee(DossierPivot dossier, GarantiePivot garantieContrat) {
+		return dossier.getContrat().getInformations().getGaranties().stream()
+				.filter(g -> g.getCodeCombinaisonFils() != null && g.getCodeCombinaisonFils().getCodePlan().equals(garantieContrat.getCodeCombinaisonFils().getCodePlan())).findFirst().orElse(null);
+	}
+
+	private BigDecimal getMontantTotal(GarantiePivot garantieContrat, GarantiePivot garantieDemandeAvenant) {
+		List<CouverturePivot> couvertures = montantAssureDemandeAvenantEditeeInferieurAMontantAsurreContrat(garantieContrat, garantieDemandeAvenant) ? garantieDemandeAvenant.getCouvertures()
+				: garantieContrat.getCouvertures();
+		return couvertures.stream().filter(nePasPrendreLesCouverturesRefusees()).map(c -> c.getMontantAssure() != null ? c.getMontantAssure() : BigDecimal.ZERO).reduce(BigDecimal.ZERO,
+				BigDecimal::add);
+	}
+
+	private boolean montantAssureDemandeAvenantEditeeInferieurAMontantAsurreContrat(GarantiePivot garantieContrat, GarantiePivot garantieDemandeAvenant) {
+		List<CouverturePivot> couverturesContrat = garantieContrat.getCouvertures();
+		List<CouverturePivot> couverturesDemandeAvenant = garantieDemandeAvenant.getCouvertures();
+
+		if (!couverturesContrat.isEmpty() && !couverturesDemandeAvenant.isEmpty()) {
+			BigDecimal montantAssureContrat = couverturesContrat.get(0).getMontantAssure();
+			BigDecimal montantAssureDemandeAvenant = couverturesDemandeAvenant.get(0).getMontantAssure();
+
+			if (montantAssureContrat != null && montantAssureDemandeAvenant != null) {
+				return montantAssureDemandeAvenant.compareTo(montantAssureContrat) < 0;
+			}
+		}
+
+		return false;
+	}
+
+	public void suspendreDemandeAvenant(final BigInteger idDemandeAvenant) {
+		sDemandeAvenant.suspendreDemandeAvenant(idDemandeAvenant);
+	}
+
+	public void enregistrerTarif(Optional<DemandeAvenantPivot> demandeAvenant) {
+		sDemandeAvenant.enregistrerTarif(demandeAvenant);
+	}
+
+	public Optional<Boolean> enregistrerNumeroTravail(BigInteger idDemandeAvenant, String numeroTravail) {
```

Comment:
Input validation issue - No validation for null idDemandeAvenant. (Severity: low)